src/resources/ResourceSnapshots/ResourceSnapshots.ts

```diff
@@ -45,19 +45,52 @@ export default class ResourceSnapshots extends Resource {
         return await fetch(url, {method: 'get'});
     }
 
-    createFromFile(file: File, options: CreateFromFileOptions) {
-        const computedOptions = {developerNotes: options.developerNotes, snapshotFileType: undefined};
+    /**
+     * Creates a snapshot from a file containing the configuration.
+     *
+     * @param {File} file The file containing the configuration.
+     * @param {CreateFromFileOptions} options
+     */
+    createFromFile(file: File, options: CreateFromFileOptions): Promise<ResourceSnapshotsModel>;
+
+    /**
+     * Creates a snapshot from a file buffer containing the configuration.
+     *
+     * @param {Buffer} file The file containing the configuration.
+     * @param {ResourceSnapshotSupportedFileTypes} fileType The type of the file containing the configuration.
+     * @param {CreateFromFileOptions} options
+     */
+    createFromFile(
+        file: Buffer,
+        fileType: ResourceSnapshotSupportedFileTypes,
+        options: CreateFromFileOptions
+    ): Promise<ResourceSnapshotsModel>;
+
+    createFromFile(
+        file: File | Buffer,
+        typeOrOptions: ResourceSnapshotSupportedFileTypes | CreateFromFileOptions,
+        options?: CreateFromFileOptions
+    ) {
+        let fileContent: File | string;
+        let computedOptions = {developerNotes: undefined, snapshotFileType: undefined};
 
-        if (file.type === 'application/zip') {
-            computedOptions.snapshotFileType = ResourceSnapshotSupportedFileTypes.ZIP;
-        } else if (file.type === 'application/json') {
-            computedOptions.snapshotFileType = ResourceSnapshotSupportedFileTypes.JSON;
+        if (Buffer.isBuffer(file)) {
+            fileContent = file.toString();
+            computedOptions = {developerNotes: options.developerNotes, snapshotFileType: typeOrOptions};
         } else {
-            throw new Error('The uploaded file must be either a ZIP or a JSON file.');
+            const type = file.type;
+            computedOptions.developerNotes = (typeOrOptions as CreateFromFileOptions).developerNotes;
+            if (type === 'application/zip') {
+                computedOptions.snapshotFileType = ResourceSnapshotSupportedFileTypes.ZIP;
+            } else if (type === 'application/json') {
+                computedOptions.snapshotFileType = ResourceSnapshotSupportedFileTypes.JSON;
+            } else {
+                throw new Error('The uploaded file must be either a ZIP or a JSON file.');
+            }
```

Comment:
I would extract that in a separate method `getTypeFromFile`, to keep the main method tidy, and/or maybe also use a switch and a more strongly typed string. (Maybe a MIME_TYPE enum or smthg 🤔 )